### DISCUSSION TITLE
No space left on device

### DISCUSSION THREAD

**Discussion by PanGan21** [fluvio-community/fluvio#4031] [Q&A] [answered]
Hi am trying to build fluvio from source in order to contribute and keep getting `No space left on device` error while building it even after cleaning some space.
I am wondering if there are any specific hardware requirements and also if there is an alternative for local development.

**Comment by sehz** [comment]
Hi,  what OS are you using for development?

**Comment by PanGan21** [comment]
Hi! Thanks for responding!
I am using: MacBook Pro (Retina, 15-inch, Mid 2015) with 16 GB memory

**Comment by sehz** [comment, marked as answer]
If you stick with using `local cluster`,  disk space should be minimum.  
To clean up disk space, you can do `cargo clean`.

To build local clusters, you just need to perform:
```
make build-cli; make build-cluster
```
On my Mac, it consumes about `6g` for this.   

I see you are using old Intel Mac which is very slow.  Try to switch to Apple silicon Machines.
